### PR TITLE
Birthday

### PR DESCRIPTION
**Pull request by Mole1424**
Im sorry Mat......
(needs databse upgrade and config change to include lord chanellor id)
will add ability to plot age later

### CODE REVIEW

File: docs.md [hunk 1]
```diff
@@ -63,9 +63,9 @@ Cogs are how Discord.py organizes commands, each file is a separate cog with sep
 - To create a DB migration:
   1. Create model in `/models`
   2. Import in `/models/__init__.py`
-  3. Run `alembic revision --autogenerate -m "<change description>"`
+  3. Run `docker compose exec apollo alembic revision --autogenerate -m "<change description>"`
   4. Check the newly created upgrade and downgrade is correct
-  5. Upgrade your database with `alembic upgrade head`
+  5. Upgrade your database with `docker compose exec apollo alembic upgrade head`
```
Comment (Joeyh021) [inline docs.md:68]: these changes are incorrect: you can't generate a migrationf from within the database unless you then copy it out afterwards. Leave these as is.

File: models/birthday.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+from sqlalchemy.orm import Mapped, mapped_column
+
+from models.models import Base
+
+
+class Birthday(Base):
+    __tablename__ = "birthdays"
+    date: Mapped[datetime] = mapped_column(primary_key=True)
+    age: Mapped[int] = mapped_column(primary_key=True)
```
Comment (Joeyh021) [inline models/birthday.py:11]: do you need to store the age here, or can you just count the number of birthdays?
Also may be worth adding a column for which user did the happy birthday message.

File: cogs/commands/birthday.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            self.age = 0
+
+    @commands.hybrid_command(help=LONG_HELP_TEXT, brief=SHORT_HELP_TEXT)
+    async def birthday(self, ctx: Context):
+        current_date = utc.localize(datetime.now()).astimezone(
+            timezone("Europe/London")
+        )
+        if current_date.date() <= self.date.date():
+            return await ctx.reply(
+                "I'm sorry but our lord chancellor has already been wished a happy birthday today"
```
Comment (Joeyh021) [inline cogs/commands/birthday.py:40]: **L**ord **C**hancellor, put some respect on the name 😤

File: README.md [hunk 4]
```diff
@@ -37,8 +37,9 @@ When you first create the database container, you'll need to apply migrations:
 1. Run `docker compose up -d` to start both services in the background. The python one won't work, but leave it running.
    - When changing any source files, the container will have to be rebuilt: `docker compose up --build`
    - For containers running in the background, you can get the outputs using `docker compose logs`
-2. Run `docker compose exec apollo alembic upgrade head` to apply the database migrations.
-3. Run `docker compose down && docker compose up` to restart both services with the migrations applied.
+2. Run `docker compose exec apollo alembic revision --autogenerate -m "<explanation>"` to add the new database
```
Comment (Joeyh021) [inline README.md:40]: this is incorrect and does not apply for just the usecase of quickstart, which the readme intends to cover.

**Review by Joeyh021 — CHANGES_REQUESTED** [review]

**Comment by Joeyh021** [comment]
You need to commit the generated migration. Ensure that what alembic generates is actually correct: it's not intended to be an entirely automated process.

File: models/birthday.py (hunk 2 above)
Comment (Mole1424) [reply models/birthday.py:11]: spoke with owen about this in tech-team, just makes life slightly easier. Good idea for the user.

File: cogs/commands/birthday.py (hunk 3 above)
Comment (Mole1424) [reply cogs/commands/birthday.py:40]: "I'm sorry but our high and gracious Lord Chancellor has already been wished a happy birthday today" - this better?

**Comment by Joeyh021** [comment]
LGTM. Tidied up the docs myself just to make it slightly more accurate. 
Feel free to merge yourself :)

**Pull request merged**